service/service.go

```diff
@@ -0,0 +1,312 @@
+                        svc.HandleError(err, message.Offset, &message.Value)
+                        continue
+                    }
+
+                    //Create GetPayment payment session URL
+                    getPaymentURL := svc.PaymentsAPIURL + "/payments/" + pp.ResourceURI
+                    log.Info("Payment URL : " + getPaymentURL)
+                    //Call GetPayment payment session from payments API
+                    paymentResponse,statusCode, err := payment.GetPayment(getPaymentURL, svc.Client, svc.ApiKey)
+                    if err != nil {
+                        log.Error(err, log.Data{"message_offset": message.Offset})
+                        svc.HandleError(err, message.Offset, &paymentResponse)
+                    }
+                    log.Info("Payment Response : ", log.Data{"payment_response": paymentResponse, "status_code":statusCode})
+                    //GetPayment Cost from payment session Cost array
+                    cost, err := paymentResponse.GetCost("cic-report")
+                    if err != nil {
+                        log.Error(err, log.Data{"message_offset": message.Offset})
+                        svc.HandleError(err, message.Offset, &paymentResponse)
+                    }
+                    log.Info("Cost : ", log.Data{"cost": cost})
+                    productCode := svc.ProductMap.Codes[cost.ProductType]
+
+                    //Create GetPayment payment URL
+                    getPaymentDetailsURL := svc.PaymentsAPIURL + "/private/payments/" + pp.ResourceURI + "/payment-details"
+                    log.Info("Payment Details URL : " + getPaymentDetailsURL)
+                    //Call GetPayment payment details from payments API
+                    paymentDetails,statusCode, err := payment.GetPaymentDetails(getPaymentDetailsURL, svc.Client, svc.ApiKey)
+                    if err != nil {
+                        log.Error(err, log.Data{"message_offset": message.Offset})
+                        svc.HandleError(err, message.Offset, &paymentDetails)
+                    }
+                    log.Info("Payment Details Response : ", log.Data{"payment_details": paymentDetails,"status_code":statusCode})
+
+                    //Build Eshu database object
+                    eshu := models.EshuResourceDao{
+                        PaymentRef:    paymentDetails.PaymentID,
+                        ProductCode:   productCode,
+                        CompanyNumber: paymentResponse.CompanyNumber,
+                        FilingDate:    "",
+                        MadeUpdate:    "",
+                    }
+
+                    //Add Eshu object to the Database
+                    err = svc.DAO.CreateEshuResource(&eshu)
+                    if err != nil {
+                        log.Error(err, log.Data{"message": "failed to create eshu request in database",
+                            "data":       eshu,
+                            "collection": svc.TranCollection})
```

Comment:
The collection isn't relevant here - the service shouldn't have knowledge of which collection we're committing data to, even if it's just for logging